test/Microsoft.VisualStudio.Composition.Tests/Reflection/TypeRefTests.cs

```diff
@@ -21,6 +21,15 @@ public void EqualsDistinguishesArrays()
             Assert.NotEqual(TypeRef.Get(typeof(object), TestUtilities.Resolver), TypeRef.Get(typeof(object[]), TestUtilities.Resolver));
         }
 
+        [Fact]
+        public void AssemblyNameInstancesReused()
+        {
+            var typeRef1 = TypeRef.Get(typeof(object), TestUtilities.Resolver);
```

Comment:
Added a really simple test to ensure this works and doesn't regress out of the blue.